### DISCUSSION TITLE
Repetir codigos HTML

### DISCUSSION THREAD

**Discussion by octavioviana** [frontendbr/forum#570] [Discussão]
Boa dia, Pessoal. É o o seguinte, gostaria de saber como eu evito de repetir os mesmos códigos em HTML todas as paginas, pois o header e o footer são os mesmos, para deixaria o código mais legível e fácil a manutenção, eu sei que poderia ou deveira? usar PHP o "Include", mas teria como usar JavaScript puro? Ou uma outra solução, algo que deve pesquisar. Desde já agradeço!!!

**Comment by kivervinicius** [comment]
Ola, uma ideia é trabalhar com algum template engine, por exemplo o handlebars, é uma solução bem elegante. http://handlebarsjs.com/, uma outra forma seria chamar estes outros conteúdos que você deseja via ajax, mas não recomendo pois a página será mais demorada para renderizar.

**Comment by kvnol** [comment]
Olá, já falamos sobre isso semana passada no tópico #560.

**Comment by octavioviana** [comment]
Obg galera... agradeço!

**Comment by mahenrique94** [comment]
@octavioviana porque você não quer utilizar o include do PHP sendo que o mesmo foi feito exatamente para isso ?

**Comment by cristofersousa** [comment]
@octavioviana, 

Conforme o  @kivervinicius  comentou, adote template engine que ele irá fazer esse trabalho braçal para você.

Dê uma olhada no [Pug - Template Engine](https://pugjs.org/) ele segue a mesma proposta do Handlebars.

Mais links de referencias:

1. https://medium.freecodecamp.com/how-to-modularize-html-using-template-engines-and-gulp-d1cb8af54138
2.  https://blog.codecasts.com.br/ecossistema-javascript-parte-06-template-engines-b41a41178dce?gi=a4a03a976ed5

[ ],s

**Comment by felipefialho** [comment]
Na issue #560 tem várias dicas sobre isso :)

**Comment by leandroruel** [comment]
jekyll
